src/docfx/lib/html/HtmlReader.cs

```diff
@@ -7,16 +7,17 @@
 namespace Microsoft.Docs.Build
 {
     /// <summary>
-    /// Provides a high-performance, near-zero-allocation, fault-tolerant, standard-compliant API
+    /// Provides a high-performance, fault-tolerant, standard-compliant API
     /// for forward-only, read-only access to HTML text.
     /// </summary>
-    internal ref struct HtmlReader
+    internal class HtmlReader
     {
         private string _html;
         private int _length;
         private int _position;
 
         private HtmlTokenType _type;
+        private bool _isSelfClosing;
```

Comment:
can just use auto property

Reply:
I intentionally use private fields here to ensure everything is a field access.